server/src/test/java/org/elasticsearch/search/profile/query/QueryProfilerTests.java

```diff
@@ -117,6 +118,49 @@ public void testBasic() throws IOException {
         assertThat(rewriteTime, greaterThan(0L));
     }
 
+    public void testConstantScoreQuery() throws IOException {
+        QueryProfiler profiler = new QueryProfiler();
+        searcher.setProfiler(profiler);
+        Query query = new ConstantScoreQuery(new TermQuery(new Term("foo", "bar")));
+        searcher.search(query, 1);
+        List<ProfileResult> results = profiler.getTree();
+        assertEquals(1, results.size());
+        Map<String, Long> breakdownConstantScoreQuery = results.get(0).getTimeBreakdown();
+        assertEquals(1, results.get(0).getProfiledChildren().size());
+        Map<String, Long> breakdownTermQuery = results.get(0).getProfiledChildren().get(0).getTimeBreakdown();
+
+        assertThat(breakdownConstantScoreQuery.get(QueryTimingType.CREATE_WEIGHT.toString()).longValue(), greaterThan(0L));
+        assertThat(breakdownConstantScoreQuery.get(QueryTimingType.BUILD_SCORER.toString()).longValue(), greaterThan(0L));
+        assertThat(breakdownConstantScoreQuery.get(QueryTimingType.NEXT_DOC.toString()).longValue(), equalTo(0L));
+        assertThat(breakdownConstantScoreQuery.get(QueryTimingType.ADVANCE.toString()).longValue(), equalTo(0L));
+        assertThat(breakdownConstantScoreQuery.get(QueryTimingType.SCORE.toString()).longValue(), greaterThan(0L));
+        assertThat(breakdownConstantScoreQuery.get(QueryTimingType.MATCH.toString()).longValue(), equalTo(0L));
+
+        assertThat(breakdownConstantScoreQuery.get(QueryTimingType.CREATE_WEIGHT.toString() + "_count").longValue(), greaterThan(0L));
+        assertThat(breakdownConstantScoreQuery.get(QueryTimingType.BUILD_SCORER.toString() + "_count").longValue(), greaterThan(0L));
+        assertThat(breakdownConstantScoreQuery.get(QueryTimingType.NEXT_DOC.toString() + "_count").longValue(), equalTo(0L));
```

Comment:
actually it shouldn't be 0 but the same value as the wrapped query since timings are supposed to include timings of children

Reply:
Thanks Adrien! I see what you mean, somewhat the constant score query should inherit the timers from their children.

Reply:
Exactly, or in that particular case, from their child since constant-score queries can only have one child.

Reply:
I think we don't want to do anything if the underlying scorer is cached since it will be consumed differently from this scorer. We should only try to reuse timing information if the underlying scorer is a ProfileScorer too?